### PR TITLE
add optional xentoollog linker flag and update opam file

### PR DESCRIPTION
**Pull request by mseri**
This uniforms the `opam` file with the one in `xs-opam` and allows to compile `xenopsd` on systems with `xen >= 4.7` where linking to `libxentoollog` is required.

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

### CODE REVIEW

File: configure.ml [hunk 1]
```diff
@@ -80,16 +80,46 @@ let expand start finish input output =
     exit 1;
   end
 
+let output_file filename lines =
+  let oc = open_out filename in
+  let lines = List.map (fun line -> line ^ "\n") lines in
+  List.iter (output_string oc) lines;
```
Comment (lindig) [inline configure.ml]: Or `List.iter (fun line -> Printf.fprintf oc "%s\n" line) lines`

File: configure.ml [hunk 2]
```diff
@@ -80,16 +80,46 @@ let expand start finish input output =
...
+  close_out oc
+
+let cc c_program =
+  let c_file = Filename.temp_file "configure" ".c" in
+  let o_file = c_file ^ ".o" in
+  output_file c_file c_program;
+  let found = Sys.command (Printf.sprintf "cc -Werror -c %s -o %s %s" c_file o_file "2>/dev/null") = 0 in
```
Comment (lindig) [inline configure.ml]: `Printf.sprintf "cc -Werror -c %s -o %s 2>/dev/null" c_file o_file` - not sure it is better but more standard.

File: configure.ml [hunk 3]
```diff
@@ -80,16 +80,46 @@ let expand start finish input output =
...
+  if Sys.file_exists c_file then Sys.remove c_file;
+  if Sys.file_exists o_file then Sys.remove o_file;
+  found
+
+let check_domain_create_has_config_param () =
+  let c_program = [
+    "#include <stdlib.h>";
+    "#include <xenctrl.h>";
+    "#include <xenguest.h>";
+    "int main(int argc, const char *argv){";
```
Comment (lindig) [inline configure.ml]: The type for `argv` is wrong. It should be `char *argv[]` or `char **argv`. I don't think it should be declared const. You don't need the `int r` - I'd just call the function and ignore the return value or cast to `(void)`.

http://stackoverflow.com/questions/779910/should-i-use-char-argv-or-char-argv-in-c

File: configure.ml [hunk 4]
```diff
@@ -80,16 +80,46 @@ let expand start finish input output =
...
+
 let configure bindir sbindir libexecdir scriptsdir etcdir mandir =
   let xenctrl = find_ocamlfind false "xenctrl" in
   let xenlight = find_ocamlfind false "xenlight" in
   let libvirt = find_ocamlfind false "libvirt" in
   let xen45 = find_seriallist () in
+  let xen_4_7 = check_domain_create_has_config_param () in
 
-  Printf.printf "Configuring with:\n\tbindir=%s\n\tsbindir=%s\n\tlibexecdir=%s\n\tscriptsdir=%s\n\tetcdir=%s\n\tmandir=%s\n\txenctrl=%b\n\txenlight=%b\n\tlibvirt=%b\n\n" bindir sbindir libexecdir scriptsdir etcdir mandir xenctrl xenlight libvirt;
+  Printf.printf "Configuring with:\n\tbindir=%s\n\tsbindir=%s\n\tlibexecdir=%s\n\tscriptsdir=%s\n\tetcdir=%s\n\tmandir=%s\n\txenctrl=%b\n\txenlight=%b\n\tlibvirt=%b\n\txentoollog=%b\n\n" bindir sbindir libexecdir scriptsdir etcdir mandir xenctrl xenlight libvirt xen_4_7;
```
Comment (lindig) [inline configure.ml]: I'd break up that line:

```
let p = Printf.sprintf in
List.iter print_endline
[ "Configure with"
; p "\tbindir=%s"   bindir
; p "\tsbindir=%s"  sbindir
; ...
]
```

**Comment by mseri** [comment]
This was a 2 minutes port of the relevant line fromxen-api-lowlevel-libs. I focused only on making it compile on my laptop. I am happy to implement the improvements that you suggest, thanks!

File: configure.ml (hunk 3 above)
Comment (mseri) [inline configure.ml]: In this case I did not really care about the detail, I copied the code from xen-api-lowlevel-libs. Do you think it is worth to cleanup the configure.ml and polish the c also there?

File: configure.ml (hunk 2 above)
Comment (mseri) [inline configure.ml]: Well spotted. i thought I had changed it, it is a leftover of deopping the verbose flag :-) I will fix it

File: configure.ml (hunk 1 above)
Comment (mseri) [inline configure.ml]: I prefer your version 👍

File: configure.ml (hunk 3 above)
Comment (lindig) [inline configure.ml]: I understand that the code will work but I think `char *argv` is too obviously wrong. It's a matter of pride ;)

**Comment by mseri** [comment]
Unfortunately the build fails because Travis pulls the old xcp-idl from our archives

**Comment by gaborigloi** [comment]
That's because the [script uploading RPMs](https://github.com/jonludlam/various-scripts) uses the last successful build.

**Comment by mseri** [comment]
We discussed it in person. Merging

**Pull request merged**